### ISSUE TITLE
Feature Request: early return from a subdir() call

### ISSUE THREAD

**Issue by rhd** [mesonbuild/meson#2779]
Hi,

This has happened a couple of times to me. Sometimes I want to partially process a `meson.build` inside of a subdir and currently you have to use big `if` blocks to get this to happen.

Example:

Top level meson.build.

```meson
if meson.is_cross()
  subdir('foo')
end
```

But if there is an include path I always need from `foo`, I can't do this. I have two choice.

1:

```meson
foo_inc = include_directories('foo/include')
if meson.is_cross()
  subdir('foo')
end
```

or 2:

`meson.build` (always include `foo`):

```meson
subdir('foo')
```
inside of `foo/meson.build`

```meson
foo_inc = include_directories('include')
if meson.is_cross()
    <rest of build definition>
    if some_other_condition:
        <more build definition>
    end
end
```

### Proposal:

At the top level, I'd always like to include `foo`

```meson
subdir('foo')
```

Inside of `foo/meson.build` it would be nice to have something like:

```meson
foo_inc = include_directories('include')
if meson.is_cross()
  return
end

subdir('bar')

if some_other_condition
  return
end

<rest of build definition>
```

Now we have the ability to partially process a build definition with any number of exit points possible without cascading `if` blocks.

Thoughts?

**Comment by behlec** [comment]
I would suggest to use `exit` instead of `return`. Return might be useful for functions.

Should it be possible to exit with some sort of error?

**Comment by tp-m** [comment]
I would not mind such functionality either, but have never been bothered enough by the `if` `endif` blocks.

Perfect syntax for me would be something like  `skip if $condition`.

Not sure about `exit`, it sounds like it should terminate processing entirely.

**Comment by behlec** [comment]
I implemented a quick first try, see #2808.

**Comment by rhd** [comment]
Looks like this was implemented by #2808 - I'm going to close this. Thank you!
